core/src/main/java/apoc/result/VirtualPath.java

```diff
@@ -0,0 +1,121 @@
+package apoc.result;
+
+import org.neo4j.graphdb.Entity;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Path;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.traversal.Paths;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.collections4.IterableUtils.reversedIterable;
+
+
+public class VirtualPath implements Path {
+
+    private final Node start;
+    private final List<Relationship> relationships = new ArrayList<>();
+
+    public VirtualPath(Node start) {
+        Objects.requireNonNull(start);
+        this.start = start;
+    }
+
+    public void addRel(Relationship relationship) {
+        Objects.requireNonNull(relationship);
+        this.relationships.add(relationship);
+    }
```

Comment:
optional: maybe add a check to ensure the rel is either connected to the last end node or to the path's start node? Would ensure we don't get disconnected paths.